gapic/templates/%namespace/%name_%version/%sub/services/%service/async_client.py.j2

```diff
@@ -388,6 +391,12 @@ class {{ service.async_client_name }}:
         )
         {% endif %}
 
+        {% if service.version %}
```

Comment:
It would be good if we could call this code in the `_client_macros.j2` file, so we reduce duplication. Maybe a separate macro call `set_version_header(service.version)`, and the macro could take care of the conditional.

Reply:
Fixed in https://github.com/googleapis/gapic-generator-python/pull/1999/commits/009fc7109ea8624d43fd7593c504ad9d494188db